01_fib/nicholascross/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	Fib(7)
+}
+
+func Fib(n int) {
+	if n == 0 {
+		return
+	}
+
+	sign := sign(n)
+	sequenceLength := sign * n
+
+	lastNumber := 0
+	currentNumber := 1
+	fmt.Fprint(out, currentNumber)
+
+	for index := 0; index < sequenceLength-1; index++ {
+		nextNumber := lastNumber + currentNumber*sign
+		lastNumber = currentNumber
+		currentNumber = nextNumber
+		fmt.Fprint(out, " ", currentNumber)
+	}
+}
+
+func sign(n int) (sign int) {
+	if n < 0 {
```

Comment:
Is a seperate function really necessary for this? Personally I don't think it is. It would be quicker to parse the `Fib` func in your head if the logic of determining sign were right there rather than off in a seperate func. I know this is a trivial implementation but sometimes it makes more sense to keep the logic for a function in that fiunction rather than splitting it out to a helper